return 500 when push a txt file as oras

Env:
OS: Ubuntu 1804
Build: harbor-offline-installer-v2.0.0-build.3158.tgz
Storage: Minio
https registry: yes

Followed the steps of oras doc.
```
oras login 10.xx.xx.xx

echo "hello world" > artifact.txt

oras push 10.xx.xx.xx/hello-artifact:v1 \
--manifest-config /dev/null:application/vnd.acme.rocket.config.v1+json \
./artifact.txt
```

It returns a 500 err
```
Uploading 85b128b7adf9 artifact.txt
Error: failed commit on ref "unknown-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855": unexpected status: 500 Internal Server Error
```

core logs shows
```
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020/03/24 08:53:05.323 #033[1;44m[D]#033[0m [transaction.go:62]  |   10.78.100.51|#033[97;43m 401 #033[0m|   9.922726ms|   match|#033[97;46m POST    #033[0m /v2/library/hello-artifact/blobs/uploads/   r:/v2/*/blobs/uploads
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020-03-24T08:53:05Z [INFO] [/server/middleware/middleware.go:57][action="request" middleware="quota" requestID="181ed3fc-2a30-4545-80ee-20233dc30b74" url="/v2/library/hello-artifact/blobs/uploads/"]: no resources request for this http request, so direct to next handler
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020/03/24 08:53:05.421 #033[1;44m[D]#033[0m [transaction.go:62]  |   10.78.100.51|#033[97;42m 202 #033[0m|  79.385292ms|   match|#033[97;46m POST    #033[0m /v2/library/hello-artifact/blobs/uploads/   r:/v2/*/blobs/uploads
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020-03-24T08:53:05Z [ERROR] [/server/middleware/quota/quota.go:91][action="request" middleware="quota" requestID="4f955278-bb35-4c6c-b68a-c777cf2823b2" url="/v2/library/hello-artifact/blobs/uploads/7d65378a-1fa0-46de-8721-1f273b28e586"]: get blob size failed, error: redigo: nil returned
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020-03-24T08:53:05Z [ERROR] [/server/middleware/middleware.go:57][action="request" middleware="quota" requestID="4f955278-bb35-4c6c-b68a-c777cf2823b2" url="/v2/library/hello-artifact/blobs/uploads/7d65378a-1fa0-46de-8721-1f273b28e586"]: get resources failed, error: redigo: nil returned
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020-03-24T08:53:05Z [ERROR] [/server/error/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"redigo: nil returned"}]}
Mar 24 08:53:05 172.26.0.1 core[1330]: 2020/03/24 08:53:05.442 #033[1;44m[D]#033[0m [transaction.go:62]  |   10.78.100.51|#033[97;41m 500 #033[0m|   6.673739ms|   match|#033[97;43m PUT     #033[0m /v2/library/hello-artifact/blobs/uploads/7d65378a-1fa0-46de-8721-1f273b28e586   r:/v2/*/blobs/uploads/:session_id
```

--- a/src/controller/blob/controller.go
+++ b/src/controller/blob/controller.go
@@ -305,6 +305,10 @@ func (c *controller) GetAcceptedBlobSize(sessionID string) (int64, error) {
 	key := fmt.Sprintf("upload:%s:size", sessionID)
 	size, err := redis.Int64(conn.Do("GET", key))
 	if err != nil {
+		if err == redis.ErrNil {
+			return 0, nil
+		}
+
 		return 0, err
 	}
 

--- a/src/controller/blob/controller_test.go
+++ b/src/controller/blob/controller_test.go
@@ -257,7 +257,8 @@ func (suite *ControllerTestSuite) TestGetSetAcceptedBlobSize() {
 	sessionID := uuid.New().String()
 
 	size, err := Ctl.GetAcceptedBlobSize(sessionID)
-	suite.NotNil(err)
+	suite.Nil(err)
+	suite.Equal(int64(0), size)
 
 	suite.Nil(Ctl.SetAcceptedBlobSize(sessionID, 100))
 

--- a/src/server/middleware/quota/put_blob_upload.go
+++ b/src/server/middleware/quota/put_blob_upload.go
@@ -44,6 +44,11 @@ func putBlobUploadResources(r *http.Request, reference, referenceID string) (typ
 		return nil, err
 	}
 
+	if size == 0 {
+		logger.Debug("blob size is 0")
+		return nil, nil
+	}
+
 	projectID, _ := strconv.ParseInt(referenceID, 10, 64)
 
 	digest := r.URL.Query().Get("digest")

--- a/src/server/middleware/quota/put_blob_upload_test.go
+++ b/src/server/middleware/quota/put_blob_upload_test.go
@@ -15,8 +15,10 @@
 package quota
 
 import (
+	"fmt"
 	"net/http"
 	"net/http/httptest"
+	"strconv"
 	"testing"
 
 	"github.com/goharbor/harbor/src/pkg/types"
@@ -26,46 +28,86 @@ import (
 
 type PutBlobUploadMiddlewareTestSuite struct {
 	RequestMiddlewareTestSuite
+
+	handler http.Handler
 }
 
-func (suite *PutBlobUploadMiddlewareTestSuite) TestMiddleware() {
-	mock.OnAnything(suite.quotaController, "IsEnabled").Return(true, nil)
+func (suite *PutBlobUploadMiddlewareTestSuite) SetupTest() {
+	suite.RequestMiddlewareTestSuite.SetupTest()
 
 	next := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
 
+	suite.handler = PutBlobUploadMiddleware()(next)
+
+	mock.OnAnything(suite.quotaController, "IsEnabled").Return(true, nil)
+}
+
+func (suite *PutBlobUploadMiddlewareTestSuite) makeRequest(contentLength int) *http.Request {
 	url := "/v2/library/photon/blobs/uploads/cbabe458-28a1-4e1b-ad15-0cb0229df4e8?digest=sha256:57c2ec3bf82f09c94be2e5c5beb124b86fcbb42e76fb82c99066c054422010e4"
+	req := httptest.NewRequest(http.MethodPut, url, nil)
+	req.Header.Set("Content-Length", strconv.Itoa(contentLength))
 
-	{
-		mock.OnAnything(suite.blobController, "Exist").Return(true, nil).Once()
+	return req
+}
 
-		req := httptest.NewRequest(http.MethodPut, url, nil)
-		req.Header.Set("Content-Length", "100")
-		rr := httptest.NewRecorder()
+func (suite *PutBlobUploadMiddlewareTestSuite) TestBlobSizeIsZero() {
+	mock.OnAnything(suite.blobController, "GetAcceptedBlobSize").Return(int64(0), nil)
 
-		PutBlobUploadMiddleware()(next).ServeHTTP(rr, req)
-		suite.Equal(http.StatusOK, rr.Code)
-	}
+	req := suite.makeRequest(0)
+	rr := httptest.NewRecorder()
 
-	{
-		mock.OnAnything(suite.blobController, "Exist").Return(false, nil).Once()
-		mock.OnAnything(suite.quotaController, "Request").Return(nil).Once().Run(func(args mock.Arguments) {
-			resources := args.Get(3).(types.ResourceList)
-			suite.Len(resources, 1)
-			suite.Equal(resources[types.ResourceStorage], int64(100))
+	suite.handler.ServeHTTP(rr, req)
+	suite.Equal(http.StatusOK, rr.Code)
+}
 
-			f := args.Get(4).(func() error)
-			f()
-		})
+func (suite *PutBlobUploadMiddlewareTestSuite) TestGetAcceptedBlobSizeFailed() {
+	mock.OnAnything(suite.blobController, "GetAcceptedBlobSize").Return(int64(0), fmt.Errorf("error"))
 
-		req := httptest.NewRequest(http.MethodPut, url, nil)
-		req.Header.Set("Content-Length", "100")
-		rr := httptest.NewRecorder()
+	req := suite.makeRequest(0)
+	rr := httptest.NewRecorder()
 
-		PutBlobUploadMiddleware()(next).ServeHTTP(rr, req)
-		suite.Equal(http.StatusOK, rr.Code)
-	}
+	suite.handler.ServeHTTP(rr, req)
+	suite.Equal(http.StatusInternalServerError, rr.Code)
+}
+
+func (suite *PutBlobUploadMiddlewareTestSuite) TestBlobExist() {
+	mock.OnAnything(suite.blobController, "Exist").Return(true, nil)
+
+	req := suite.makeRequest(100)
+	rr := httptest.NewRecorder()
+
+	suite.handler.ServeHTTP(rr, req)
+	suite.Equal(http.StatusOK, rr.Code)
+}
+
+func (suite *PutBlobUploadMiddlewareTestSuite) TestBlobNotExist() {
+	mock.OnAnything(suite.blobController, "Exist").Return(false, nil).Once()
+	mock.OnAnything(suite.quotaController, "Request").Return(nil).Once().Run(func(args mock.Arguments) {
+		resources := args.Get(3).(types.ResourceList)
+		suite.Len(resources, 1)
+		suite.Equal(resources[types.ResourceStorage], int64(100))
+
+		f := args.Get(4).(func() error)
+		f()
+	})
+
+	req := suite.makeRequest(100)
+	rr := httptest.NewRecorder()
+
+	suite.handler.ServeHTTP(rr, req)
+	suite.Equal(http.StatusOK, rr.Code)
+}
+
+func (suite *PutBlobUploadMiddlewareTestSuite) TestBlobExistFailed() {
+	mock.OnAnything(suite.blobController, "Exist").Return(false, fmt.Errorf("error"))
+
+	req := suite.makeRequest(100)
+	rr := httptest.NewRecorder()
+
+	suite.handler.ServeHTTP(rr, req)
+	suite.Equal(http.StatusInternalServerError, rr.Code)
 }
 
 func TestPutBlobUploadMiddlewareTestSuite(t *testing.T) {

--- a/src/server/middleware/quota/quota.go
+++ b/src/server/middleware/quota/quota.go
@@ -83,7 +83,7 @@ func RequestMiddleware(config RequestConfig, skippers ...middleware.Skipper) fun
 
 		if !enabled {
 			// quota is disabled for the reference object, so direct to next handler
-			logger.Infof("quota is disabled for %s %s, so direct to next handler", reference, referenceID)
+			logger.Debugf("quota is disabled for %s %s, so direct to next handler", reference, referenceID)
 			next.ServeHTTP(w, r)
 			return
 		}
@@ -98,7 +98,7 @@ func RequestMiddleware(config RequestConfig, skippers ...middleware.Skipper) fun
 
 		if len(resources) == 0 {
 			// no resources request for this http request, so direct to next handler
-			logger.Info("no resources request for this http request, so direct to next handler")
+			logger.Debug("no resources request for this http request, so direct to next handler")
 			next.ServeHTTP(w, r)
 			return
 		}
@@ -206,7 +206,7 @@ func RefreshMiddleware(config RefreshConfig, skipers ...middleware.Skipper) func
 		}
 
 		if !enabled {
-			logger.Infof("quota is disabled for %s %s, so return directly", reference, referenceID)
+			logger.Debugf("quota is disabled for %s %s, so return directly", reference, referenceID)
 			return nil
 		}
 

